### PR TITLE
feat: add client-project-setup package

### PR DESCRIPTION
**Pull request by alaincormier-ssc**
new package for issue #370
(additional PRs will be created for other packages and to update documentation)

**`solutions/client-project-setup`**
- includes / deprecates packages:
    - none
- adds features:
    - client tier3 and tier4 namespaces
    - client tier3/4 root sync
    - client shared vpc service project
- depends on:
    - solutions/client-landing-zone

### CODE REVIEW

**Review by lucstjean-ssc — DISMISSED** [review]
/lgtm

**Review by cartyc — DISMISSED** [review]
/lgtm

File: solutions/client-project-setup/namespaces/project-id-tier3.yaml [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+metadata:
+  name: configconnectorcontext.core.cnrm.cloud.google.com
+  namespace: project-id-tier3 # kpt-set: ${project-id}-tier3
+  annotations:
+    cnrm.cloud.google.com/ignore-clusterless: "true"
+    config.kubernetes.io/depends-on: resourcemanager.cnrm.cloud.google.com/namespaces/client-name-projects/Project/project-id # kpt-set: resourcemanager.cnrm.cloud.google.com/namespaces/${client-name}-projects/Project/${project-id}
+spec:
+  googleServiceAccount: tier3-sa@project-id.iam.gserviceaccount.com # kpt-set: tier3-sa@${project-id}.iam.gserviceaccount.com
+---
+# Give KCC, for this namespace, permission to read KCC resources in the networking namespace.
```
[the remark this answers is not in the document]
Comment (alaincormier-ssc) [reply solutions/client-project-setup/namespaces/project-id-tier3.yaml]: removed

File: solutions/client-project-setup/namespaces/project-id-tier4.yaml [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMPolicyMember
+metadata:
+  name: project-id-tier4-sa-networkuser-host-project-id-permissions # kpt-set: ${project-id}-tier4-sa-networkuser-${host-project-id}-permissions
+  namespace: client-name-projects # kpt-set: ${client-name}-projects
+  annotations:
+    cnrm.cloud.google.com/ignore-clusterless: "true"
+    config.kubernetes.io/depends-on: resourcemanager.cnrm.cloud.google.com/namespaces/client-name-projects/Project/project-id # kpt-set: resourcemanager.cnrm.cloud.google.com/namespaces/${client-name}-projects/Project/${project-id}
+spec:
+  resourceRef:
```
[the remark this answers is not in the document]
Comment (alaincormier-ssc) [reply solutions/client-project-setup/namespaces/project-id-tier4.yaml:60]: todo note added

**Review by davelanglois-ssc — APPROVED** [review]

**Review by lucstjean-ssc — APPROVED** [review]
/lgtm and Nice!

**Review by cartyc — APPROVED** [review]

**Pull request merged**